### PR TITLE
perf(app): paginate /evals server-side results

### PR DESCRIPTION
**Pull request by will-holley**
This refactors the /evals table to use server-side pagination instead of loading every eval summary in one request. The GET /api/results route now validates optional `limit` and `offset` query params and returns pagination metadata when pagination is requested while preserving the existing unpaginated response shape. The eval summary model supports paginated retrieval plus total-count queries, and the frontend /evals table renders through the shared DataTable's server-virtualized row mode.

## Keeping global search working

Server-virtualized DataTables hid the global search box outright (`showGlobalFilterControl = !isServerVirtualized`), so moving /evals behind server pagination would have silently dropped a search control that `main` had just fixed and regression-tested. Column filters already had a `manualFiltering` escape hatch; global search had none.

- `DataTable` gains the matching `manualGlobalFiltering` opt-in with controlled `globalFilter` / `onGlobalFilterChange`. In that mode the client row model is left unfiltered and the owner applies the term itself.
- Because the client only holds a window of rows, the term has to reach the query: `GET /api/results` gains a `search` param that filters on the eval id and description with `LIKE`, wildcards escaped so `%` matches literally. Both the page query and the total count honor it, as does the unpaginated response shape.
- `EvalsTable` debounces the term (200ms), threads it through every fetch (first page, virtualized range loads, and CSV export), and resets the virtualization cache so results start from row 0.

Also drops the unused duplicate `GetResultsQuerySchema` that was added alongside the pagination params; `ServerSchemas.ResultList.Query` is the schema the route actually uses.

## Test Plan
- `npm run test:app -- --run` (272 files / 5310 tests) — includes new `DataTable` coverage for the `manualGlobalFiltering` escape hatch and the updated `EvalsTable.search.test.tsx`, whose fixture now applies search the way the server does.
- `npx vitest run test/models/eval.test.ts test/server test/types` — new coverage for search by description and by id, null descriptions, literal `%`/`_` wildcards, search-aware counts/pagination, and whitespace-only terms.
- `npx tsc --noEmit -p tsconfig.json`, `(cd src/app && npx tsc -b)`, `npm run build`, `npm run architecture:check`, `npm run knip`, Biome.
- Live-checked against a local DB of 5192 evals: searching `sequence doc` narrows 5192 rows to the one match (proving it searches the whole table, not the loaded page), `search=%` returns 0 rather than everything, deep scrolling still pages in rows, and clearing the box restores the full list.

### CODE REVIEW

**Comment by ghost** [comment]
**No tests generated**

[View output ↗](https://app.usetusk.ai/app/testing-commit-check/38c601ef-2baf-4432-b45d-e47e1d1c66fa?client=e91e3433-8fb4-4e98-b233-bba492f4909c)

No symbols were selected for unit test generation because all meaningful logic changes in `src/app/pages/evals/components/EvalsTable.tsx` are already comprehensively covered by the existing test suite (`EvalsTable.test.tsx`). The remaining code consists of simple memoizations and rendering logic, which are either trivial or better suited for integration tests. There are no untested business logic paths or edge cases warranting further unit tests.

---
<details><summary>View check history</summary>
<p>

| Commit | Status | Output | Created (UTC) |
|--------|--------|--------|--------|
| 7cc703651f6e6732c35f00112483888b97adc71c | Generated 13 tests - 13 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/1d17b6f7-d955-4cad-a15b-6b386e4d1ffc?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Feb 26, 2026 7:18PM |
| fd8cec25a8b9f989bcabaa46e2d998d0ece73f7e | Generated 13 tests - 13 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/a370d125-24d7-4e1d-b6a6-9e2e11ead297?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Feb 26, 2026 8:01PM |
| b93ded8e314c9aa4f7f72e72bdeb0d54cf73ff24 | No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/38c601ef-2baf-4432-b45d-e47e1d1c66fa?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Mar 5, 2026 9:32PM |

</p>
</details>

File: src/server/server.ts [hunk 1]
```diff
@@ -146,25 +171,37 @@ export function createApp() {
...
-      const previousResults = await getEvalSummaries(
-        req.query.datasetId,
-        req.query.type,
-        req.query.includeProviders,
-      );
-      res.json({ data: previousResults });
-    },
-  );
+  app.get('/api/results', async (req: Request, res: Response): Promise<void> => {
+    const queryResult = ResultsQuerySchema.safeParse(req.query);
```
Comment (mldangelo) [inline src/server/server.ts]: this is great, can you ask your agent to put to organize the schema with the other route schemas?

File: src/app/src/pages/evals/components/EvalsTable.tsx [hunk 2]
```diff
@@ -46,30 +57,48 @@ export default function EvalsTable({
...
+      } finally {
+        if (!signal.aborted) {
+          setIsLoading(false);
+        }
       }
-    }
-  }, []);
+    },
+    [pagination.pageIndex, pagination.pageSize, useServerPagination],
+  );
```
[the remark this answers is not in the document]
Comment (mldangelo) [reply src/app/src/pages/evals/components/EvalsTable.tsx:201]: is this valid?

**Review by mldangelo — COMMENTED** [review]
looks pretty good, thank you

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo** [comment]
Closing this. Main solved the problem it targeted in a different way, and the pagination approach now costs more than it saves.

**The load cost is already fixed on main**
- #8537 (Apr 20) virtualized the shared `DataTable`, so `/evals` renders only the rows on screen.
- #9462 (May 26) stopped reading and deserializing eval `config` JSON on the summary path. It measured `/api/results` at 27,495 evals going from 1,350 ms to 233 ms. It also chose client virtualization over pagination on purpose, because search, CSV export and delete all rely on having the full row set.

**QA today against a local DB with 5,197 evals**

| | `main` (all rows) | This branch (`limit=50`) |
| --- | --- | --- |
| `GET /api/results` (warm) | ~80 ms, 242 KB gzipped | ~3 ms, 2 KB |
| Browser fetch + JSON parse | 78–80 ms | n/a |
| Sort/filter controls in `/evals` column headers | 12 (sort + filter on all 6 columns) | **0** |

**What the branch costs**
- Once the table switches to server pagination, every column sets `enableSorting: false` and the column filters are hidden, because `/api/results` has no server-side sort or filter. So the page saves about 75 ms and loses sorting and filtering on every column.
- Restoring those would mean adding server sort and filter params on top of the new `limit`/`offset`/`search`/count API, the `manualGlobalFiltering` mode added to `DataTable`, and the duplicated `DataTable` render branch in `EvalsTable`. That is already about 900 lines for roughly 75 ms at 5k evals.

If an installation ever gets big enough for this to matter again, the better route is the one `ModelAuditHistory` uses: `useServerVirtualizedRows` with `useDataTableServerSorting` and server-side filters, so sorting and filtering keep working.

Thanks @will-holley for pushing on this.
